math/binary_exponent.cpp

```diff
@@ -19,53 +19,92 @@
  * \f}
  * Hence to calculate 2^10 we only need to multiply \f$2^8\f$ and \f$2^2\f$
  * skipping \f$2^1\f$ and \f$2^4\f$.
+ * 
+ * Bitwise Operations Used:
+ * a>>1 => is same as a/2.
+ * a>>=1 => is same as a = a/2.
+ * a&1 => is same as a%2. Checks wether number a is odd or not. Results to 1 i.e. True if a is odd and 0 or False otherwise.
+ *
+ * Modular Properties
+ * ( a * a ) % m = ( a%m * a%m ) % m
+ * Using this property of modular aithmetic we can efficiently calculate (a^b)%m. 
  */
 
 #include <iostream>
 
 /// Recursive function to calculate exponent in \f$O(\log(n))\f$ using binary
 /// exponent.
-int binExpo(int a, int b) {
+uint64_t binExpo(uint64_t a, uint64_t b) {
     if (b == 0) {
         return 1;
     }
-    int res = binExpo(a, b / 2);
-    if (b % 2) {
+    uint64_t res = binExpo(a, b>>1); 
+    if (b&1) { // Same as b%2==1. Result to true if b is odd.
         return res * res * a;
     } else {
         return res * res;
     }
 }
 
+/// Recursive function to calculate exponent modular m (a^b %m) in \f$O(\log(n))\f$ using 
+/// binary exponentiation.
+uint64_t binModExpo(uint64_t a, uint64_t b, uint64_t m) {
```

Comment:
Function documentation is not up to the [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html).